pkg/linter/lintRuleSyntacticRedundantCommon.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package linter
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+const NetworkACL = "network ACL" // todo: https://github.com/np-guard/vpc-network-config-analyzer/issues/724
+
+// a rule that is implied syntactically redundant:
+// nacl rule - implied by higher priority rules in the nacl table
+// sg rule - implied by other rules in the sg table
+type ruleRedundant struct {
+	rule         vpcmodel.RuleOfFilter
+	containRules map[int]*vpcmodel.RuleOfFilter // indexes of rules because of which this rule is redundant to their description
+	vpcResource  vpcmodel.VPC
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////
+// functionality used by both SG and NACL lints
+////////////////////////////////////////////////////////////////////////////////////////////
+
+// Rule is syntactically redundant in SG if other rules in the table implies it
+// Rule is syntactically redundant in NACL if it is shadowed by higher priority rules of the table
+// A rule is shadowed in NACL if the rule will never determine the actual allow/deny status
+// (of any connection in any configuration)
+// A rule in a table has 3 dimensions: source, destination and connection
+// SG rule is redundant if the 3-dimensions union of the other rules contains it, thus rule is implied by others
+// NACL rule  is redundant if the 3-dimensions union of the higher priority (allow or deny) rules contains it
+// The above 3 dimensions containment is checked as follows:
+// in the below a "point" is a point in the src x dst x conn 3-dimensions
+// If each point in the rule is contained in the others/higher priority rules then it is redundant
+// This we check as follows:
+// For each table:
+//  1. partition the IPBlock to maximal atomic blocks w.r.t. the table
+//  2. For each rule r:
+//     2.1. Iterate over the cartesian product of r's src and dst atomic blocks
+//     For each item <src_atomic, dst_atomic>
+//     compute the union of the connection of the other/higher priority rules whose <src, dst> contains <src_atomic, dst_atomic>
+//     If the resulting conn does not contain the connection of the rule r then the block is not redundant
+//     Done? Rule is redundant
+//
+//nolint:gocyclo // better not split into two function
+func findRuleSyntacticRedundant(configs map[string]*vpcmodel.VPCConfig,
+	filterLayerName string) (res []ruleRedundant, err error) {
+	for _, config := range configs {
+		if config.IsMultipleVPCsConfig {
+			continue // no use in executing lint on dummy vpcs
+		}
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
move GetFilterTrafficResourceOfKind() and GetRules() into getTableOrientedStructs()

Reply:
done